src/test/miner_tests.cpp

```diff
@@ -187,6 +187,39 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
       // printf("%lu %lu <= %lu\n", (long unsigned int) blockSize, (long unsigned int) pblocktemplate->block.nBlockSize, (long unsigned int) maxGeneratedBlock);
       delete pblocktemplate;
     }
+
+    coinbaseReserve.value=0;  // Test no reserve
+     // Now generate lots of full size blocks and verify that none exceed the maxGeneratedBlock value
+    for (unsigned int i = 2000; i <= 80000; i+=37)
+    {
+      maxGeneratedBlock = i;
+
+      pblocktemplate = CreateNewBlock(chainparams, scriptPubKey);
+      BOOST_CHECK(pblocktemplate);
+      BOOST_CHECK(pblocktemplate->block.fExcessive == false);
+      BOOST_CHECK(pblocktemplate->block.nBlockSize <= maxGeneratedBlock);
+      unsigned int blockSize = pblocktemplate->block.GetSerializeSize(SER_NETWORK, CTransaction::CURRENT_VERSION);
+      BOOST_CHECK(blockSize <= maxGeneratedBlock);
+      //printf("%lu %lu <= %lu\n", (long unsigned int) blockSize, (long unsigned int) pblocktemplate->block.nBlockSize, (long unsigned int) maxGeneratedBlock);
+      delete pblocktemplate;
+    }
+
+    std::string testMinerComment("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLM");
+     // Now generate lots of full size blocks and verify that none exceed the maxGeneratedBlock value
+    //printf("test mining with different sized miner comments");
```

Comment:
Actually on point number 1 I'm not right, these are just blocktemplates so the txns never get mined.  So that looks fine.